packages/core/test/persistence.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { expect, describe, it, beforeEach, afterEach } from "vitest";
+import { getDefaultLoggerOptions, pino } from "@walletconnect/logger";
+import { ICore, IStore } from "@walletconnect/types";
+import { TEST_CORE_OPTIONS, disconnectSocket, throttle } from "./shared";
+import { Core, Store } from "../src";
+import { generateRandomBytes32 } from "@walletconnect/utils";
+
+let core: ICore;
+let store: IStore<any, any>;
+const n_restarts = 1; // number of restarts to use for persistence tests
+
+const logger = pino(getDefaultLoggerOptions({ level: "fatal" }));
+type MockValue = { id: string; value: string };
+const MOCK_STORE_NAME = "persistent-store";
+const storeTestValues = [
+  { id: "1", value: "foo" },
+  { id: "2", value: "bar" },
+  { id: "3", value: "baz" },
+];
+
+const initCore = async (dbName = "persistent-test") => {
+  const coreOptions = {
+    ...TEST_CORE_OPTIONS,
+    storageOptions: { database: `tmp/${dbName}.db` },
+  };
+  core = new Core(coreOptions);
+  await core.start();
+};
+
+const initStore = async () => {
+  store = new Store<string, MockValue>(
+    core,
+    logger,
+    MOCK_STORE_NAME,
+    undefined,
+    (val) => val.value,
+  );
+  await store.init();
+  storeTestValues.forEach((val) => store.set(val.id, val));
+};
+
+/**
+ *  Prevents gross code duplication in tests that require restarting core
+ * @param fx function to run before each restart
+ * @param fy function to run after each restart
+ */
+const restartCore = async (fx?: () => Promise<void>, fy?: () => Promise<void>) => {
+  for (let i = 0; i < n_restarts; i++) {
+    if (fx) await fx();
+    await initCore();
+    if (fy) await fy();
+  }
+};
```

Comment:
```suggestion
const restartCore = async (beforeStartRoutine?: () => Promise<void>, afterStartRoutine?: () => Promise<void>) => {
  for (let i = 0; i < n_restarts; i++) {
    if (beforeStartRoutine) await beforeStartRoutine();
    await initCore();
    if (afterStartRoutine) await afterStartRoutine();
  }
};
```
Feel free to be extra verbose in namings as `fx`/`fy` are not really descriptive unless digging into params/usage to gauge intent